pkg/scale/decode.go

```diff
@@ -426,6 +428,34 @@ func (ds *decodeState) decodeArray(dstv reflect.Value) (err error) {
 	return
 }
 
+func (ds *decodeState) decodeMap(dstv reflect.Value) (err error) {
+	l, err := ds.decodeLength()
+	if err != nil {
+		return
+	}
+	in := dstv.Interface()
+
+	for i := uint(0); i < l; i++ {
+		tempKeyType := reflect.TypeOf(in).Key()
+		tempKey := reflect.New(tempKeyType).Elem()
+		err = ds.unmarshal(tempKey)
+		if err != nil {
+			return
+		}
+
+		tempElemType := reflect.TypeOf(in).Elem()
+		tempElem := reflect.New(tempElemType).Elem()
+		err = ds.unmarshal(tempElem)
+		if err != nil {
+			return
+		}
+
+		dstv.SetMapIndex(tempKey, tempElem)
+	}
+
+	return
```

Comment:
nit to be explicit / the function is not that short

```suggestion
	return nil
```